src/Sharpbrake.Client/NoticeBuilder.cs

```diff
@@ -30,27 +31,50 @@ public NoticeBuilder()
         }
 
         /// <summary>
-        /// Sets exception data into notice errors list.
+        /// Sets error entries into notice errors list.
         /// </summary>
-        public void SetErrorEntries(Exception exception)
+        public void SetErrorEntries(Exception exception, string message)
         {
             notice.Exception = exception;
+            notice.Message = message;
 
             var errors = new List<ErrorEntry>();
+            var ex = exception;
 
-            // main exception + no more than 3 inner exceptions (to reduce JSON size)
-            while (errors.Count < 4 && exception != null)
+            if (ex != null)
             {
-                var exceptionType = exception.GetType();
                 errors.Add(new ErrorEntry
                 {
-                    Message = !string.IsNullOrEmpty(exception.Message)
-                        ? string.Format("{0}: {1}", exceptionType.Name, exception.Message)
-                        : exceptionType.Name,
-                    Type = exceptionType.FullName,
-                    Backtrace = Utils.GetBacktrace(exception)
+                    Message = !string.IsNullOrEmpty(message) ? message : ex.Message,
+                    Type = ex.GetType().FullName,
+                    Backtrace = Utils.GetBacktrace(new StackTrace(ex, true))
                 });
-                exception = exception.InnerException;
+
+                ex = ex.InnerException;
+            }
+            else
+            {
+                errors.Add(new ErrorEntry
+                {
+                    Message = message,
+#if !NETSTANDARD1_4
```

Comment:
Just curious, did NETSTANDARD1_4 not have backtraces (or the StackTrace class)?

Reply:
.NET Standard 1.x doesn't implement a lot of APIs available in the standard .NET Framework (so-called "Desktop"), most of them were added later - in [.NET Standard 2.x](https://raw.githubusercontent.com/dotnet/standard/master/docs/versions/netstandard2.0_diff.md) (search for "StackTrace"). For 1.x branch some APIs were implemented in separate packages, for StackTrace it's "System.Diagnostics.StackTrace" (version 4.3.0) - but it adds only the variant with Exception in its constructor that doesn't help as we don't have Exception in that case.